nixos/modules/services/web-apps/akkoma.md

```diff
@@ -0,0 +1,136 @@
+{option}`_secret` – a string pointing to the file containing the actual value of the option.
+
+For the mandatory configuration settings these secrets will be generated automatically if the
+referenced file does not exist during startup, unless disabled through
+{option}`services.akkoma.initSecrets`.
+
+The following configuration binds Akkoma to the local IPv6 address `::1` on port 4000, expecting
+to be run behind a HTTP proxy on `fediverse.example.com`.
+
+
+```nix
+services.akkoma.enable = true;
+services.akkoma.config = {
+  ":pleroma" = {
+    ":instance" = {
+      name = "My Akkoma instance";
+      description = "More detailed description";
+      email = "admin@example.com";
+      registration_open = false;
+    };
+
+    "Pleroma.Web.Endpoint" = {
+      url.host = "fediverse.example.com";
+      http = {
+        ip = "::1";
+        port = 4000;
+      };
+    };
+  };
+};
+```
+
+Please refer to the [configuration cheat sheet](https://docs.akkoma.dev/stable/configuration/cheatsheet/)
+for additional configuration options.
+
+The service can be enabled with the usual
+
+```ShellSession
+$ nixos-rebuild switch
+```
+
+## User management {#modules-services-akkoma-user-management}
+
+After the Akkoma service is running, the administration utility can be used to
+[manage users](https://docs.akkoma.dev/stable/administration/CLI_tasks/user/). In particular an
+administrative user can be created with
+
+```ShellSession
+$ sudo -u akkoma RELEASE_COOKIE=/run/akkoma/COOKIE \
+pleroma_ctl user new <nickname> <email> --admin --moderator --password <password>
+```
+
+## Proxy configuration {#modules-services-akkoma-proxy-configuration}
+
+Although it is possible to expose Akkoma directly, it is common practice to operate it behind an
+HTTP reverse proxy such as HAProxy or nginx.
+
+```nix
+security.acme = {
+  email = "admin@example.com";
+  acceptTerms = true;
+};
+
+services.akkoma.nginx = {
+  enableACME = true;
+  forceSSL = true;
+};
+
+services.nginx = {
+  enable = true;
+
+  clientMaxBodySize = "16m";
+  recommendedTlsSettings = true;
+  recommendedOptimisation = true;
+  recommendedGzipSettings = true;
+};
+```
+
+## Frontend management {#modules-services-akkoma-frontend-management}
+
+Akkoma will be deployed with the `pleroma-fe` and `admin-fe` frontends by default. These can be
+overridden by setting the {option}`services.akkoma.frontends` option.
+
+The following example overrides the primary frontend’s default configuration using a custom
+derivation.
+
+```nix
+services.akkoma.frontends.primary.package = let
+  default-config = pkgs.writeText "default-config.json" (builtins.toJSON {
+    expertLevel = 1;
+    collapseMessageWithSubject = false;
+    stopGifs = false;
+    replyVisibility = "following";
+    webPushHideIfCW = true;
+    hideScopeNotice = true;
+    renderMisskeyMarkdown = false;
+    hideSiteFavicon = true;
+    postContentType = "text/markdown";
+    showNavShortcuts = false;
+  });;
```

Comment:
Is there any advantage to first translating the json to nix instead of using json direct? Also why do we write this to the store first? We could directly inject that into jq.

Reply:
I copied this from another user without giving it too much thought. But I agree that writing the JSON to the store first is unnecessary – I suspect that the original author was unfamiliar with `passAsFile`. I personally feel that writing an attribute set is somewhat more comfortable than a string of JSON.